### PR TITLE
systemd: fix build on centos

### LINKED ISSUE #7658: systemd: build error
- [ x ] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [ x ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [ x ] ran `brew update` and can still reproduce the problem?
- [ x ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ x ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

I need `systemd` as a dependency for upgrading `octave`.

`brew install -v systemd 2>&1`

failed with the message

```
In file included from ./src/basic/util.h:46:0,
                 from ./src/basic/hashmap.h:29,
                 from ./src/shared/bus-util.h:30,
                 from src/core/socket.c:33:
./src/basic/missing.h:494:17: error: expected identifier before numeric constant
 #define MS_MOVE 8192
                 ^
Makefile:18031: recipe for target 'src/core/libcore_la-socket.lo' failed
make[2]: *** [src/core/libcore_la-socket.lo] Error 1
Makefile:24263: recipe for target 'install-recursive' failed
make[1]: *** [install-recursive] Error 1
Makefile:25843: recipe for target 'install' failed
make: *** [install] Error 2
```

Full logs are [here](https://gist.github.com/0d57d92cb038416e2da0e5beb83ec171)

### PR DESCRIPTION
**Pull request by dpo**
fixes #7658

- [X] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by dpo** [comment]
Not sure what's going on on TravisCI. It doesn't seem to be doing anything productive.

As far as I can tell, the CircleCI build is successful; I don't see any error in the log even though it's mysteriously reported as a failure.

File: Formula/systemd.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@ class Systemd < Formula
   homepage "https://wiki.freedesktop.org/www/Software/systemd/"
   url "https://github.com/systemd/systemd/archive/v234.tar.gz"
   sha256 "da3e69d10aa1c983d33833372ad4929037b411ac421fb085c8cee79ae1d80b6a"
+  revision 3
```
Comment (iMichka) [inline Formula/systemd.rb]: This should be: `revision OS.mac? ? 2 : 3`

File: Formula/systemd.rb [hunk 2]
```diff
@@ -27,6 +27,11 @@ class Systemd < Formula
   depends_on "util-linux" # for libmount
   depends_on "libgpg-error" => :build
 
+  patch do
+    url "https://github.com/systemd/systemd/commit/227b8a762fea1458547be2cdf0e6e4aac0079730.diff?full_index=1"
+    sha256 "3a123a6cba8cf1e27b2ace04f81c6d0a87e29c7a9900aa2ec1d1fea8e06656a8"
+  end
```
Comment (iMichka) [inline Formula/systemd.rb]: The patch is linux only, so please change this to `end unless OS.mac?`

**Comment by iMichka** [comment]
Regarding the OS.mac? thing, it is just a way for me to to know what to do when merging from upstream homebrew-core to this repo. It is not perfect but works quite well.

**Comment by dpo** [comment]
Should I have kept the empty `bottle` block?

**Comment by iMichka** [comment]
@dpo Thanks! :)

**Pull request closed without merging**